enzyme/test/Enzyme/ReverseMode/cblas_ddot_double.ll

```diff
@@ -0,0 +1,191 @@
+;RUN: if [ %llvmver -ge 8 ]; then %opt < %s %loadEnzyme -enzyme -S | FileCheck %s; fi
+
+;#include <cblas.h>
+
+;extern double __enzyme_autodiff(void *, double *, double *, double *,
+;                                 double *);
+
+;double g(double *m, double *n) {
+;  double x = cblas_ddot(3, m, 1, n, 1);
+;  m[0] = 11.0;
+;  m[1] = 12.0;
+;  m[2] = 13.0;
+;  n[0] = 21.0;
+;  n[1] = 22.0;
+;  n[2] = 23.0;
+;  double y = x * x;
+;  return y;
+;}
+
+;int main() {
+;  double m[3] = {1, 2, 3};
+;  double m1[3] = {0, 0, 0};
+;  double n[3] = {4, 5, 6};
+;  double n1[3] = {0, 0, 0};
+;  double val = __enzyme_autodiff((void*)g, m, m1, n, n1);
+;  return 1;
+;}
+
+target datalayout = "e-m:e-i64:64-f80:128-n8:16:32:64-S128"
+target triple = "x86_64-unknown-linux-gnu"
+
+@__const.main.m = private unnamed_addr constant [3 x double] [double 1.000000e+00, double 2.000000e+00, double 3.000000e+00], align 16
+@__const.main.n = private unnamed_addr constant [3 x double] [double 4.000000e+00, double 5.000000e+00, double 6.000000e+00], align 16
+
+; Function Attrs: noinline nounwind optnone uwtable
+define dso_local double @g(double* %m, double* %n) {
+entry:
+  %m.addr = alloca double*, align 8
+  %n.addr = alloca double*, align 8
+  %x = alloca double, align 8
+  %y = alloca double, align 8
+  store double* %m, double** %m.addr, align 8
```

Comment:
Can you clean up the test a bit? There's no need for all of the m.addr / x, etc (e.g. run mem2reg on it).

Similarly I would run -mem2reg -instsimplify -simplifycfg after enzyme to make the gradient easier to read.